src/main/java/com/autotune/analyzer/services/UpdateResults.java

```diff
@@ -61,41 +62,28 @@ public void init(ServletConfig config) throws ServletException {
     protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
         try {
             String inputData = request.getReader().lines().collect(Collectors.joining());
-            List<JsonObject> autoTuneOperatorDataList = Arrays.asList(new Gson().fromJson(inputData, JsonObject[].class));
-            LOGGER.debug(mainKruizeExperimentMap.toString());
-            LOGGER.debug(mainKruizeExperimentMap.keySet().toString());
-
-            for (JsonObject jsonObject : autoTuneOperatorDataList) {
-                String exp_name = jsonObject.get("experiment_name").getAsString();
-                KruizeObject exp = (KruizeObject) mainKruizeExperimentMap.get(exp_name);
-                if (null == exp) {
-                    sendErrorResponse(response, null, HttpServletResponse.SC_BAD_REQUEST, "Experiment name not found");
-                } else {
-                    List<JsonObject> results = null;
-                    if (exp.getResults() == null)
-                        results = new ArrayList<>();
-                    else
-                        results = exp.getResults();
-                    results.add(jsonObject);
-                    exp.setResults(results);
-                }
+            List<ExperimentResultData> experimentResultDataList = Arrays.asList(new Gson().fromJson(inputData, ExperimentResultData[].class));
+            ActivityResultData experimentInitiator = new ExperimentInitiator().validateAndUpdateResults(mainKruizeExperimentMap, experimentResultDataList);
+            if (experimentInitiator.isSuccess()) {
+                sendSuccessResponse(response, "Results added successfully");
```

Comment:
Logging should happen inside `validateAndUpdateResults`, that way it is common in the case of both k8s native and REST API invocation